File: frontend/site_validade_medeiros/lib/model/register_model.dart
class RegisterModel {
  String? id;
  String? ean;
  String? description;
  String? validity;
  int? quantity;
  String? shop;
  String? zone;
  String? corridor;
  String? registrationDate;

  RegisterModel(
      {this.id,
      this.ean,
      this.description,
      this.validity,
      this.quantity,
      this.shop,
      this.zone,
      this.corridor,
      this.registrationDate});

  RegisterModel.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    ean = json['ean'];
    description = json['description'];
    validity = json['validity'];
    quantity = json['quantity'];
    shop = json['shop'];
    zone = json['zone'];
    corridor = json['corridor'];
    registrationDate = json['registrationDate'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['ean'] = ean;
    data['description'] = description;
    data['validity'] = validity;
    data['quantity'] = quantity;
    data['shop'] = shop;
    data['zone'] = zone;
    data['corridor'] = corridor;
    data['registrationDate'] = registrationDate;
    return data;
  }
}

File: backend/brigada-validade/src/main/java/com/carloslima/brigadavalidade/models/RegisterModel.java
package com.carloslima.brigadavalidade.models;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.Date;
import java.util.UUID;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

@Entity
@Table(name = "TB_VALIDADE")
public class RegisterModel implements Serializable{
	private static final long serialVersionUID = 1L;
	
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private UUID id;
	@Column(nullable = false, unique = true, length = 13)
	private String ean;
	@Column(nullable = false)
	private String description;
	@Column(nullable = false)
	@Temporal(TemporalType.TIMESTAMP)
	private Date validity;
	@Column(nullable = false)
	private int quantity;
	@Column(nullable = false, length = 7)
	private String shop;
	@Column(nullable = false, length = 10)
	private String zone;
	@Column(length = 3)
	private String corridor;
	@Column(nullable = false)
	private LocalDateTime registrationDate;
	
	public UUID getId() {
		return id;
	}
	public void setId(UUID id) {
		this.id = id;
	}
	public String getEan() {
		return ean;
	}
	public void setEan(String ean) {
		this.ean = ean;
	}
	public String getDescription() {
		return description;
	}
	public void setDescription(String description) {
		this.description = description;
	}
	public Date getValidity() {
		return validity;
	}
	public void setValidity(Date validity) {
		this.validity = validity;
	}
	public Integer getQuantity() {
		return quantity;
	}
	public void setQuantity(Integer quantity) {
		this.quantity = quantity;
	}
	public String getShop() {
		return shop;
	}
	public void setShop(String shop) {
		this.shop = shop;
	}
	public String getZone() {
		return zone;
	}
	public void setZone(String zone) {
		this.zone = zone;
	}
	public String getCorridor() {
		return corridor;
	}
	public void setCorridor(String corridor) {
		this.corridor = corridor;
	}
	public LocalDateTime getRegistrationDate() {
		return registrationDate;
	}
	public void setRegistrationDate(LocalDateTime registrationDate) {
		this.registrationDate = registrationDate;
	}
	

}
